codegen/templates/tchannel_client.tmpl

```diff
@@ -411,3 +441,34 @@ type {{$clientName}} struct {
 {{end -}}
 {{end -}}
 {{end}}
+
+func isSystemError(err error) bool {
+	if err == nil {
+		return false
+	}
+	_, isSysErr := err.(tchannel.SystemError)
+	return isSysErr
+}
+
+func setContextSystemErrorCode(ctx context.Context, err error) context.Context {
+	if ctx != nil && err != nil {
+		if systemErr, ok := err.(tchannel.SystemError); ok {
+			if code, ok := _tchannelCodeToCode[systemErr.Code()]; ok {
+				ctx = setContextStatusCode(ctx, code)
+			} else {
+				// same as yarpc-go https://github.com/yarpc/yarpc-go/blob/d33ff85d687eb11de3324507ffdc817a39001b3f/transport/tchannel/error.go#L67C39-L67C51
+				ctx = setContextStatusCode(ctx, yarpcerrors.CodeInternal)
+			}
+		}
+	}
+	return ctx
+}
+
+func setContextStatusCode(ctx context.Context, code yarpcerrors.Code) context.Context {
+	if ctx != nil {
+		if statusCode := ctx.Value(_errorCodeAnnotationKey); statusCode == nil {
+			ctx = context.WithValue(ctx, _errorCodeAnnotationKey, code)
+		}
+	}
+	return ctx
+}
```

Comment:
Can you move this to runtime folder. This is common code and would just increase binary size being present in all generated code files.